### PR TITLE
exact voc_match

### PR DESCRIPTION
**Pull request by JarbasAl**
adds an option to self.voc_match in skills, TODO from https://github.com/MycroftAI/skill-playback-control/blob/20.02/__init__.py#L34

### CODE REVIEW

**Comment by devops-mycroft** [comment]
Voight Kampff Integration Test Succeeded  ([Results](https://reports.mycroft.ai/core/PR-2675))

**Comment by devops-mycroft** [comment]
Voight Kampff Integration Test Succeeded  ([Results](https://reports.mycroft.ai/core/PR-2675))

File: mycroft/skills/mycroft_skill/mycroft_skill.py [hunk 1]
```diff
@@ -585,9 +588,14 @@ def voc_match(self, utt, voc_filename, lang=None):
             vocab = read_vocab_file(voc)
             self.voc_match_cache[cache_key] = list(chain(*vocab))
         if utt:
-            # Check for matches against complete words
-            return any([re.match(r'.*\b' + i + r'\b.*', utt)
-                        for i in self.voc_match_cache[cache_key]])
+            if exact:
+                # Check for exact match
+                return utt and any(i.strip() == utt
```
Comment (krisgesling) [inline mycroft/skills/mycroft_skill/mycroft_skill.py:593]: Do we need the `utt and` given line 590 checks for the existence of `utt` already?

**Review by krisgesling — COMMENTED** [review]
Thanks Jarbas, makes sense to shift this from the Playback Skill and provide it as an option for all Skills.

Can we also add some new cases to the voc_match unit tests to cover this feature?

File: mycroft/skills/mycroft_skill/mycroft_skill.py (hunk 1 above)
Comment (JarbasAl) [reply mycroft/skills/mycroft_skill/mycroft_skill.py:593]: good catch

**Comment by devops-mycroft** [comment]
Voight Kampff Integration Test Failed ([Results](https://reports.mycroft.ai/core/PR-2675))

**Comment by devops-mycroft** [comment]
Voight Kampff Integration Test Succeeded  ([Results](https://reports.mycroft.ai/core/PR-2675))

**Comment by JarbasAl** [comment]
will re-open upon clarification of privacy ramifications of the latest mycroft partnership

code is free, you can copy and open a new PR if you do not want to wait for me re-opening it

**Pull request closed without merging**